### ISSUE TITLE
Missing feature - Array.lenght >= 0

### ISSUE THREAD

**Issue by auirarrazaval** [typescript-eslint/typescript-eslint#1831]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/<rule>": ?
  }
}
```

```TS
if (this.indices != null && this.indices.length >= 0) {
     (code....)
}
```

**Expected Result**

Should rise a warning as length is always >= to 0

**Actual Result**

No warning

**Additional Info**

I'm not sure this is the correct way to ask this, sorry if it's not

**Comment by bradzacher** [comment]
This is impossible to check.
Typescript reports an array's length property as type `number`.
Because we only know it is a `number`, we cannot ever know that it `!== 0`.
